event_manager_test.go

```diff
@@ -76,7 +77,7 @@ func TestEventManager_QueueEvent_100_Flush(t *testing.T) {
 	// Wait for raw event queue to drain
 	require.Eventually(t, func() bool {
 		queueLength, _ := c.eventQueue.internalQueue.UserQueueLength()
-		return queueLength == 10
+		return queueLength-startingLength >= 10
```

Comment:
what is the OOO here?
```suggestion
		return ((queueLength - startingLength) >= 10)
```